app/src/main/java/org/schabi/newpipe/player/resolver/NewPipeHlsPlaylistParserFactory.java

```diff
@@ -0,0 +1,47 @@
+package org.schabi.newpipe.player.resolver;
+
+import android.net.Uri;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.android.exoplayer2.source.hls.playlist.HlsMasterPlaylist;
+import com.google.android.exoplayer2.source.hls.playlist.HlsMediaPlaylist;
+import com.google.android.exoplayer2.source.hls.playlist.HlsPlaylist;
+import com.google.android.exoplayer2.source.hls.playlist.HlsPlaylistParserFactory;
+import com.google.android.exoplayer2.upstream.ParsingLoadable;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class NewPipeHlsPlaylistParserFactory implements HlsPlaylistParserFactory {
+
+    private final HlsPlaylist hlsPlaylist;
+
+    public NewPipeHlsPlaylistParserFactory(final HlsPlaylist hlsPlaylist) {
+        this.hlsPlaylist = hlsPlaylist;
+    }
+
+    private class NewPipeHlsPlayListParser implements ParsingLoadable.Parser<HlsPlaylist> {
+
+        @Override
+        public HlsPlaylist parse(final Uri uri,
+                                 final InputStream inputStream) throws IOException {
+            return hlsPlaylist;
+        }
+    }
+
+    @NonNull
+    @Override
+    public ParsingLoadable.Parser<HlsPlaylist> createPlaylistParser() {
+        return new NewPipeHlsPlayListParser();
+    }
+
+    @NonNull
+    @Override
+    public ParsingLoadable.Parser<HlsPlaylist> createPlaylistParser(
+            @NonNull final HlsMasterPlaylist masterPlaylist,
+            @Nullable final HlsMediaPlaylist previousMediaPlaylist) {
```

Comment:
(not really a problem) Why does exoplayer pass in these parameters? May they be useful?

Reply:
I don't really know myself but here is what ExoPlayer documentation says for this method (in `HlsPlaylistParserFactory` interface):

```java
  /**
   * Returns a playlist parser for playlists that were referenced by the given {@link
   * HlsMasterPlaylist}. Returned {@link HlsMediaPlaylist} instances may inherit attributes from
   * {@code masterPlaylist}.
   *
   * @param masterPlaylist The master playlist that referenced any parsed media playlists.
   * @param previousMediaPlaylist The previous media playlist or null if there is no previous media
   *     playlist.
   * @return A parser for HLS playlists.
   */
```